src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -107,132 +99,6 @@ public HypervisorResource(ConsumerResource consumerResource, ConsumerCurator con
-        if (hostGuestMap.remove("") != null) {
-            log.warn("Ignoring empty hypervisor id");
-        }
-
-        // Maps virt hypervisor ID to registered consumer for that hypervisor, should one exist:
-        VirtConsumerMap hypervisorConsumersMap =
-            consumerCurator.getHostConsumersMap(owner, hostGuestMap.keySet());
-
-        int emptyGuestIdCount = 0;
-        Set<String> allGuestIds = new HashSet<>();
-
-        Collection<List<String>> idsLists = hostGuestMap.values();
-        for (List<String> guestIds : idsLists) {
-            // ignore null guest lists
-            // See bzs 1332637, 1332635
-            if (guestIds == null) {
-                continue;
-            }
-            for (Iterator<String> guestIdsItr = guestIds.iterator(); guestIdsItr.hasNext();) {
-                String id = guestIdsItr.next();
-
-                if (StringUtils.isEmpty(id)) {
-                    emptyGuestIdCount++;
-                    guestIdsItr.remove();
-                }
-                else {
-                    allGuestIds.add(id);
-                }
-            }
-        }
-
-        if (emptyGuestIdCount > 0) {
-            log.warn("Ignoring {} empty/null guest id(s).", emptyGuestIdCount);
-        }
-
-        HypervisorUpdateResultDTO result = new HypervisorUpdateResultDTO();
-        for (Entry<String, List<String>> hostEntry : hostGuestMap.entrySet()) {
-            String hypervisorId = hostEntry.getKey();
-            // Treat null guest list as an empty list.
-            // We can get an empty list here from katello due to an update
-            // to ruby on rails.
-            // (https://github.com/rails/rails/issues/13766#issuecomment-32730270)
-            // See bzs 1332637, 1332635
-            if (hostEntry.getValue() == null) {
-                hostEntry.setValue(new ArrayList<>());
-            }
-            try {
-                log.debug("Syncing virt host: {} ({} guest IDs)", hypervisorId, hostEntry.getValue().size());
-
-                boolean hostConsumerCreated = false;
-                boolean updatedType = false;
-                // Attempt to find a consumer for the given hypervisorId
-                Consumer consumer = null;
-                if (hypervisorConsumersMap.get(hypervisorId) == null) {
-                    if (!createMissing.booleanValue()) {
-                        log.info("Unable to find hypervisor with id {} in org {}", hypervisorId, ownerKey);
-                        result.setFailedUpdate(addFailed(result.getFailedUpdate(),
-                            hypervisorId + ": " +
-                            i18n.tr("Unable to find hypervisor in org \"{0}\"", ownerKey)));
-                        continue;
-                    }
-                    log.debug("Registering new host consumer for hypervisor ID: {}", hypervisorId);
-                    consumer = createConsumerForHypervisorId(hypervisorId, owner, principal);
```

Comment:
I think we should be good to remove the `createConsumerForHypervisorId` method. It looks like the hypervisorUpdate method is the only one using that private method.